### PR TITLE
Added propane sensor support for iGrill v3

### LINKED ISSUE #12: Propane Sensor?
My igrill3 has a propane level sensor. Any chance of adding that to the project? I should add I would be happy to buy you coffee!

### PR DESCRIPTION
**Pull request by apop880**
Closes #12.

### CODE REVIEW

**Comment by bendikwa** [comment]
Thanks for contributing!
I'll try to find the time to test it soon, but it looks good. 👍

-Bendik

**Comment by ryantiger658** [comment]
Any progress on the testing? I would love this feature but don't want to leave the main branch!  

I am happy to help with testing if needed.

File: utils.py [hunk 1]
```diff
@@ -107,6 +107,8 @@ def publish(temperatures, battery, heating_element, client, base_topic, device_n
         client.publish("{0}/{1}/battery".format(base_topic, device_name), battery)
     if heating_element:
         client.publish("{0}/{1}/heating_element".format(base_topic, device_name), heating_element)
+    if propane_level:
```
Comment (bendikwa) [inline utils.py:110]: If propane_level is `0.0` the if test will fail, and no level will be transmitted.

File: igrill.py [hunk 2]
```diff
@@ -124,6 +131,9 @@ def read_battery(self):
     def read_heating_elements(self):
         return bytearray(self.heating_elements.read()) if self.has_heating_element else None
 
+    def read_propane_level(self):
+        return float(bytearray(self.propane_level.read())[0])*25 if self.has_propane_sensor else None
```
Comment (bendikwa) [inline igrill.py:135]: On my iGrill3 with no propane sensor, this returnes `0.0`

**Review by bendikwa — COMMENTED** [review]
Hi. Sorry for the long wait, I have to admit that I completely forgot about this.

I have tested a bit, and it looks like it works even if my iGrill 3 does not have a propane level (installed).
In my case the read_propane_level() return 0.0. This will in turn end up sending no message on MQTT since `0.0` is not interpreted as `true`. While this is fine for the iGrills missing the propane sensor, this will probably also mean that if the actual level sensor ends up with  returning `0.0` you will lose the messages?
Is this the intended behavior?

File: utils.py (hunk 1 above)
Comment (eth0up) [reply utils.py:110]:
```suggestion
    if propane_level is not None:
```

Comment (apop880) [reply utils.py:110]: I think that should work. Started a new job recently and just haven't had a chance to go back and test, but I believe you're right that that is the change that needs to be made @eth0up.

**Comment by bendikwa** [comment]
Closed due to inactivity

**Pull request closed without merging**